arch/arm64/core/isr_wrapper.S

```diff
@@ -32,9 +32,20 @@ GDATA(_sw_isr_table)
 GTEXT(_isr_wrapper)
 SECTION_FUNC(TEXT, _isr_wrapper)
 
-	/* ++(_kernel->nested) to be checked by arch_is_in_isr() */
-	inc_nest_counter x0, x1
-
+	/* ++_current_cpu->nested to be checked by arch_is_in_isr() */
+	get_cpu	x0
+	ldr	w1, [x0, #___cpu_t_nested_OFFSET]
+	add	w1, w1, #1
+	str	w1, [x0, #___cpu_t_nested_OFFSET]
+
+	/* If not nested: switch to IRQ stack and save current sp on it. */
+	cmp	w1, #1
```

Comment:
Hi npitre, correct me if I am wrong. I wonder if it is possible that when in a nested IRQ, there does a context switch after `daifclr`?
If does, after a context switch in a nested IRQ, it eret to a new high priority thread and now the nested counter is 1 because the first IRQ hasn't been completed yet.
At this time, a new IRQ comes, then it would not switch to IRQ stack because the nested counter is now bigger than 1.